packages/rich-text/src/component/index.js

```diff
@@ -527,6 +527,12 @@ function RichText( {
 	function handleSpace( event ) {
 		const { keyCode, shiftKey, altKey, metaKey, ctrlKey } = event;
 
+		// Allow space characters to be typed in a <summary>.
+		if ( keyCode === SPACE && TagName === 'summary' ) {
+			handleChange( insert( createRecord(), ' ' ) );
```

Comment:
I don't think RichText should be responsible for inserting characters, because this varies by keyboards and language and user settings. If the browser is blocking spaces to be inserted because of some other behaviour, the behaviour should be prevented some other way.

Reply:
Thanks for looking at this! Are you saying that the fix for this issue belongs somewhere else, or that we should be smarter about the locale of the character we are inserting on line 532? I want to do it the right way, and would appreciate your help understanding what that is.

I spent a lot of time on #23940 looking for a way to fix this at the block level but couldn't get it to work without any luck. Something about the way that Firefox handles the spacebar in a `<summary contenteditable>` element eats the space character, so the two solutions I came up with were:

1. Add this space inserter
2. Make the `RichText` a div child of the `<summary>` element, which means either rendering that on the client side or having separate CSS for the editor and client

I appreciate any pointers on how to make this better, and thanks again for the feedback.

Reply:
I don't know honestly. First, since this is a problem only with the summary element, we should fix it in the block and not in rich text imo. Maybe you can prevent default the event? Not sure. If the only way is to insert the character ourselves, maybe have a look at the data property in the event and insert that data?